src/hooks/api/useApiCompounderPoolDynamic.ts

```diff
@@ -44,12 +42,14 @@ export function useApiCompounderPoolDynamic({
 }) {
   const isCurve = useIsCurve(type)
   const { address } = useAccount()
+  const chainId = useChainId()
 
   return useQuery(
-    ["pools", isCurve ? "curve" : "balancer", "data", poolId, address],
+    [chainId, "pools", isCurve ? "curve" : "balancer", "data", poolId, address],
```

Comment:
We must include `chainId` in all of our query keys when fetching chain-specific data. Omission of this will cause the data to not be refetched when chainId changes.